app/data_model/submitted_responses.py

```diff
@@ -0,0 +1,81 @@
+import boto3
+from botocore.exceptions import BotoCoreError, ConnectionError as BotoConnectionError
+from app.settings import EQ_ENVIRONMENT_PREFIX, EQ_SUBMISSION_VALIDITY, EQ_DYNAMODB_ENDPOINT
+from structlog import get_logger
+
+logger = get_logger()
+
+
+class SubmittedResponses(object):
+
+    def __init__(self):
+        if EQ_DYNAMODB_ENDPOINT:
+            dynamoDB = boto3.resource('dynamodb', endpoint_url=EQ_DYNAMODB_ENDPOINT)
+        else:
+            dynamoDB = boto3.resource('dynamodb')
+
+        tableName = EQ_ENVIRONMENT_PREFIX + '-submitted-responses'
+        self.table = dynamoDB.Table(tableName)
+
+    def insert_item(self, tx_id, answers, metadata, submitted, time_to_live):
+        """Insert an item to table"""
+
+        if not time_to_live or time_to_live > EQ_SUBMISSION_VALIDITY:
+            time_to_live = EQ_SUBMISSION_VALIDITY
+
+        valid_until = int(submitted + time_to_live)
+
+        answers = self.clean_empty(answers)
+        metadata = self.clean_empty(metadata)
+
+        item = {
+            'tx_id': tx_id,
+            'answers': answers,
+            'metadata': metadata,
+            'valid_until': valid_until
+        }
+
+        try:
+            response = self.table.put_item(Item=item)['ResponseMetadata']['HTTPStatusCode']
+            if response == 200:
+                return True
+            else:
+                raise BotoConnectionError()
+        except (BotoCoreError, BotoConnectionError):
+            logger.bind(tx_id=tx_id)
+            logger.info('Could not load data into DynamoDB')
+            return False
+
+    def clean_empty(self, d):
+        if not isinstance(d, (dict, list)):
+            return d
+        if isinstance(d, list):
+            return [v for v in (self.clean_empty(v) for v in d) if v]
+        return {k: v for k, v in ((k, self.clean_empty(v)) for k, v in d.items()) if v}
+
+    def get_item(self, tx_id):
+        """
+        Get an item given its key
+        """
+        try:
+            response = self.table.get_item(Key={'tx_id': tx_id})
+            return response['Item']
```

Comment:
If there are no results does the 'Item' key exist?

Reply:
I've badly used Exception handling to deal with expected behaviour which I'll refactor.